src/api/v2/paths/directory.js

```diff
@@ -13,17 +13,19 @@ import { getDirectories } from '../db/ldap/directory-dao';
 const get = async (req, res) => {
   try {
     const params = req.query;
-    const paramKeys = Object.keys(params);
+    const paramKeys = _.keys(params);
+    let errors = [];
     if (paramKeys.includes('primaryAffiliation') && paramKeys.length === 3) {
-      return errorBuilder(res, 400, ['primaryAffiliation may not be used as a lone query parameter.']);
+      errors.push('primaryAffiliation may not be used as a lone query parameter.');
     }
 
     const valuelessParams = _.pickBy(params, (value) => !value && value !== 0);
     const generateErrorString = (accumulator, value, key) => {
       accumulator.push(`Query parameter '${key}' must have a value.`);
       return accumulator;
     };
-    const errors = _.reduce(valuelessParams, generateErrorString, []);
+    const emptyValueErrors = _.reduce(valuelessParams, generateErrorString, []);
+    errors = _.concat(errors, emptyValueErrors);
```

Comment:
Correct me if I'm wrong, but I think these two lines can be combined to:

```js
errors =  _.reduce(valuelessParams, generateErrorString, errors);
```

Reply:
you're right, I forgot about the third argument